### DISCUSSION TITLE
How to create 3D wind fields with Pyvista

### DISCUSSION THREAD

**Discussion by multilingualtricker** [pyvista/pyvista#1748] [Q&A] [answered]
Hi there,

I'm trying to create a 3D plot with vectors but am struggling even with the help given in 
https://docs.pyvista.org/examples/01-filter/glyphs.html#sphx-glr-examples-01-filter-glyphs-py and

where the data used to plot is a complex preexisting object. 

Can anyone give me a hint as to how this is done?
This is as far as I got!

longs = np.arange(0, 360, 3)
lats = np.arange(90, -93, -3)
levels = np.array([   1,    2,    3,    5,    7,   10,   20,   30,   50,   70,  100,  125,
        150,  175,  200,  225,  250,  300,  350,  400,  450,  500,  550,  600,
        650,  700,  750,  775,  800,  825,  850,  875,  900,  925,  950,  975,
       1000], dtype=np.int32)

import pyvista as pv
import numpy as np
x, y, z, = np.meshgrid(longs, lats, levels,)
grid = pv.StructuredGrid(x, y, z, )
grid.add_field_data(ds.u.data, 'u')
grid.add_field_data(ds.v.data, 'v')
grid.add_field_data(ds.w.data, 'w')

**Comment by RichardScottOZ** [comment]
What should it look like?

**Comment by whophil** [comment, marked as answer]
@multilingualtricker "field data" is confusingly named in VTK. Field data has nothing to do with the geometry of the data (see https://docs.pyvista.org/user-guide/data_model.html?highlight=field%20data#field-data). You should be adding your u, v, w data as either "point data" or "cell data," depending on if it is located at your grid nodes or cell centers.
